panic in runebuffer

No repro case yet:

```
panic: runtime error: slice bounds out of range [2:1]

goroutine 10 [running]:
github.com/ergochat/readline.(*runeBuffer).WriteRunes.func1()
	github.com/ergochat/readline@v0.1.2/runebuf.go:141 +0x179
github.com/ergochat/readline.(*runeBuffer).refresh(0xc000064360, 0xc00046bc00)
	github.com/ergochat/readline@v0.1.2/runebuf.go:450 +0x54
github.com/ergochat/readline.(*runeBuffer).WriteRunes(0xc000064360, {0xc0000122ec, 0x1, 0x1})
	github.com/ergochat/readline@v0.1.2/runebuf.go:140 +0x1f5
github.com/ergochat/readline.(*runeBuffer).WriteRune(...)
	github.com/ergochat/readline@v0.1.2/runebuf.go:111
github.com/ergochat/readline.(*operation).readline(0xc0000643c0, 0x0)
	github.com/ergochat/readline@v0.1.2/operation.go:349 +0x143b
github.com/ergochat/readline.(*operation).Runes(0xc0000643c0)
	github.com/ergochat/readline@v0.1.2/operation.go:447 +0x1a7
github.com/ergochat/readline.(*operation).String(...)
	github.com/ergochat/readline@v0.1.2/operation.go:409
github.com/ergochat/readline.(*Instance).ReadLine(0x1?)
	github.com/ergochat/readline@v0.1.2/readline.go:208 +0x17
github.com/ergochat/readline.(*Instance).Readline(0x0?)
	github.com/ergochat/readline@v0.1.2/readline.go:213 +0x13
main.runClient.func1()
	./ircdog.go:331 +0x82
created by main.runClient in goroutine 1
	./ircdog.go:328 +0x210
```

--- a/runebuf.go
+++ b/runebuf.go
@@ -87,9 +87,6 @@ func (r *runeBuffer) Len() int {
 
 func (r *runeBuffer) MoveToLineStart() {
 	r.Refresh(func() {
-		if r.idx == 0 {
-			return
-		}
 		r.idx = 0
 	})
 }
@@ -248,21 +245,23 @@ func (r *runeBuffer) Kill() {
 
 func (r *runeBuffer) Transpose() {
 	r.Refresh(func() {
-		if len(r.buf) == 1 {
-			r.idx++
-		}
-
-		if len(r.buf) < 2 {
+		if r.idx == 0 {
+			// match the GNU Readline behavior, Ctrl-T at the start of the line
+			// is a no-op:
 			return
 		}
 
-		if r.idx == 0 {
-			r.idx = 1
-		} else if r.idx >= len(r.buf) {
-			r.idx = len(r.buf) - 1
+		// OK, we have at least one character behind us:
+		if r.idx < len(r.buf) {
+			// swap the character in front of us with the one behind us
+			r.buf[r.idx], r.buf[r.idx-1] = r.buf[r.idx-1], r.buf[r.idx]
+			// advance the cursor
+			r.idx++
+		} else if r.idx == len(r.buf) && len(r.buf) >= 2 {
+			// swap the two characters behind us
+			r.buf[r.idx-2], r.buf[r.idx-1] = r.buf[r.idx-1], r.buf[r.idx-2]
+			// leave the cursor in place since there's nowhere to go
 		}
-		r.buf[r.idx], r.buf[r.idx-1] = r.buf[r.idx-1], r.buf[r.idx]
-		r.idx++
 	})
 }
 

